src/Microsoft.CodeAnalysis.BannedApiAnalyzers/Core/SymbolIsBannedAnalyzerBase.cs

```diff
@@ -240,6 +240,19 @@ bool VerifyType(Action<Diagnostic> reportDiagnostic, ITypeSymbol? type, SyntaxNo
                         return false;
                     }
 
+                    foreach (var currentNamespace in GetContainingNamespaces(type))
+                    {
+                        if (IsBannedSymbol(currentNamespace, out entry))
+                        {
+                            reportDiagnostic(
+                                syntaxNode.CreateDiagnostic(
+                                    SymbolIsBannedRule,
+                                    currentNamespace.ToDisplayString(),
```

Comment:
Is there a reason why the code above passes `SymbolDisplayFormat` to `type.ToDisplayString`, but we don't do so here? If not, consider extracting the entire if statement into a local function used for both type and namespace symbols.

Reply:
The `SymbolDisplayFormat` is setup to `...`, which produces the following when banning the namespace `System.Threading`:

> The symbol 'Threading' is banned in this project

Whereas I think it is more appropriate for the error to read:

> The symbol 'System.Threading' is banned in this project

I think this format is the format expressed by `SymbolDisplayFormat.CSharpErrorMessageFormat`. What do you want me to do here?

- Use `SymbolDisplayFormat`, which produces the first (not appropriate imo)
- Use `null` what I have currently, producing the latter
- Use `SymbolDisplayFormat.CSharpErrorMessageFormat` producing the same (in the test cases)

Reply:
It is fine to keep your current implementation if that was intentionally different from types .